verde/utils.py

```diff
@@ -261,6 +261,137 @@ def maxabs(*args, nan=True, percentile=100):
+        of the arrays. Defaults to 0, giving the minimum value.
+    max_percentile : float
+        Return the supplied percentile (0 to 100) instead of the maximum value
+        of the arrays. Defaults to 100, giving the maximum value.
+
+    Returns
+    -------
+    min, max : float
+        The minimum and maximum (or percentile) values across all arrays.
+
+    Examples
+    --------
+
+    >>> result = minmax((1, -10, 25, 2, 3))
+    >>> tuple(map(float, result))
+    (-10.0, 25.0)
+    >>> result = minmax(
+    ...     (1, -10.5, 25, 2), (0.1, 100, -500), (-200, -300, -0.1, -499)
+    ... )
+    >>> tuple(map(float, result))
+    (-500.0, 100.0)
+
+    If the array contains NaNs, we'll use the ``nan`` version of of the numpy
+    functions by default. You can turn this off through the *nan* argument.
+
+    >>> import numpy as np
+    >>> result = minmax((1, -10, 25, 2, 3, np.nan))
+    >>> tuple(map(float, result))
+    (-10.0, 25.0)
+    >>> result = minmax((1, -10, 25, 2, 3, np.nan), nan=False)
+    >>> tuple(map(float, result))
+    (nan, nan)
+
+    If a more robust statistic is desired, you can use ``min_percentile`` and
+    or ``max_percentile`` to get the values at given percentiles instead of
+    the minimum and maximum.
+
+    >>> import numpy as np
+    >>> result = minmax(
+    ...     (1, -10, 25, 2, 3), min_percentile=2, max_percentile=98
+    ... )
+    >>> tuple(map(float, result))
+    (-9.12, 23.24)
+    >>> result = minmax(
+    ...     (1, -10, 25, 2, 3), min_percentile=0, max_percentile=100
+    ... )
+    >>> tuple(map(float, result))
+    (-10.0, 25.0)
+
+    """
+    arrays = [np.atleast_1d(i) for i in args]
+
+    # checks
+    if not isinstance(min_percentile, (float, int)):
+        raise TypeError(
+            f"Invalid 'min_percentile' of type '{type(min_percentile).__name__}'. Must be a float or int."
+        )
+    if not isinstance(max_percentile, (float, int)):
+        raise TypeError(
+            f"Invalid 'max_percentile' of type '{type(max_percentile).__name__}'. Must be a float or int."
+        )
+    if not min_percentile <= max_percentile:
+        raise ValueError(
+            f"'min_percentile' ({min_percentile}) must not be larger than 'max_percentile' ({max_percentile})."
+        )
+    if min_percentile < 0 or min_percentile > 100:
+        raise ValueError(
+            f"Invalid value for 'min_percentile' ({min_percentile}). Must be between 0 and 100."
+        )
+    if max_percentile < 0 or max_percentile > 100:
+        raise ValueError(
+            f"Invalid value for 'max_percentile' ({max_percentile}). Must be between 0 and 100."
+        )
+
+    # determine which functions to use
+    if nan:
+        npmin = np.nanmin
+        npmax = np.nanmax
+        nppercentile = np.nanpercentile
+    elif not nan:
+        npmin = np.min
+        npmax = np.max
+        nppercentile = np.percentile
+    else:
+        raise TypeError(f"The 'nan' parameter ({nan}) must be a boolean.")
```

Comment:
Nit picking here. If we want to check if `nan` is boolean at the same time we assign the functions we'll use later on, I think we should check if `nan is True` or `nan is False`, or else raise the error. With this setup we could pass an object that has the `__bool__` method as `nan` but is not a boolean (not True or False), and the code would still work.

```suggestion
    if nan is True:
        npmin = np.nanmin
        npmax = np.nanmax
        nppercentile = np.nanpercentile
    elif nan is False:
        npmin = np.min
        npmax = np.max
        nppercentile = np.percentile
    else:
        raise TypeError(f"The 'nan' parameter ({nan}) must be a boolean.")
```

Reply:
Ok that soudns good to me. I thought `if nan` was exactly equivalent to `if nan is True`.